File: Android/VKAudioPlayer/app/src/main/java/github/y0rrrsh/vkaudioplayer/network/service/VKAPService.java
package github.y0rrrsh.vkaudioplayer.network.service;

import github.y0rrrsh.vkapi.VKApi.VKArrayCallback;
import github.y0rrrsh.vkapi.VKApi.VKCallback;
import github.y0rrrsh.vkaudioplayer.models.AudioModel;
import github.y0rrrsh.vkaudioplayer.models.FriendModel;
import github.y0rrrsh.vkaudioplayer.models.GroupModel;
import github.y0rrrsh.vkaudioplayer.models.UserModel;
import github.y0rrrsh.vkaudioplayer.models.dto.AudioDTO;
import github.y0rrrsh.vkaudioplayer.models.dto.FriendDTO;
import github.y0rrrsh.vkaudioplayer.models.dto.GroupDTO;
import github.y0rrrsh.vkaudioplayer.models.dto.UserDTO;

/**
 * @author Artur Yorsh
 */
public interface VKAPService {

    void getAudios(Integer userId, VKArrayCallback<AudioModel> callback);

    void getGroups(VKArrayCallback<GroupModel> callback);

    void getFriends(VKArrayCallback<FriendModel> callback);

    void addAudio(Integer id, Integer ownerId, VKCallback<Integer> callback);

    void removeAudio(Integer id, Integer ownerId, VKCallback<Integer> callback);

    void restoreAudio(Integer id, Integer ownerId, VKCallback<AudioModel> callback);

    void getUserInfo(Integer id, VKCallback<UserModel> callback);
}

File: iOS/VkAudio/network/VKAPService.swift
//
//  VKAP.swift
//  VkAudio
//
//  Created by mac-224 on 24.06.16.
//  Copyright © 2016 y0rrrsh. All rights reserved.
//

import Foundation
import UIKit

protocol VKAPService {    
    func getAudios(ownerId: Int?, callback: VKApiCallback<[AudioModel]>?)
    func getGroups(callback: VKApiCallback<[GroupModel]>?)
    func getFriends(callback: VKApiCallback<[FriendModel]>?)
    func getUserInfo(userId: Int, callback: VKApiCallback<UserModel>?)
    
    func addAudio(audioId: Int, ownerId: Int, callback: VKApiCallback<Int>?)
    func removeAudio(audioId: Int, ownerId: Int, callback: VKApiCallback<Int>?)
    func restoreAudio(id: Int, ownerId: Int, callback: VKApiCallback<AudioModel>?)
}